extensions/spring-web/runtime/src/main/java/io/quarkus/spring/web/runtime/ResponseEntityConverter.java

```diff
@@ -22,7 +22,7 @@ public static Response toResponse(ResponseEntity responseEntity) {
     private static Headers<Object> toHeaders(HttpHeaders springHeaders) {
         Headers<Object> jaxRsHeaders = new Headers<>();
         for (Map.Entry<String, List<String>> entry : springHeaders.entrySet()) {
-            jaxRsHeaders.addAll(entry.getKey(), entry.getValue());
+            jaxRsHeaders.addAll(entry.getKey(), entry.getValue().toArray(new Object[0]));
```

Comment:
Maybe let's properly initialize the size of the array to avoid creating a new one?

Reply:
Not sure I follow. We just need to convert a list to an array and according to http://shipilev.net/blog/2016/arrays-wisdom-ancients/#_conclusion this is the proper way to do it.
Am I missing something?

Reply:
I don't know. The List javadoc clearly pushes towards using the size but I suppose it's not critical enough here to have us waste our time discussing it :).

Reply:
Yeah, this isn't on a critical path anyway so it doesn't really matter.